src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -4526,10 +4526,16 @@ object Desugar extends LazyLogging {
 
       def goE(expr: PExpression): Writer[in.Node] = expr match {
         case p: PInvoke => info.resolve(p) match {
-          case Some(x: ap.PredicateCall) => predicateCallAccD(ctx, info)(x)(src)
+          case Some(x: ap.PredicateCall) =>
+            // we turn a `PredicateAccess` into an `Access` to unify this case
+            // with desugaring a PAccess:
+            for {
+              pa <- predicateCallAccD(ctx, info)(x)(src)
+            } yield in.Access(in.Accessible.Predicate(pa), in.FullPerm(pa.info))(pa.info)
```

Comment:
Is there a reason why this is preferable to what we had before?

Reply:
So far, we only allowed predicate calls but no access as termination measure. Thus, it did not matter whether we store the desugared predicate call in the internal AST and defer forming the access to the encoding phase or whether we form the access already in the desugarer. However, the latter has the advantage we treat a predicate call and an access uniformly as an access in the internal AST.